### PR TITLE
Terminate execution for criu that does not meet version requirements

### PR DESCRIPTION
**Pull request by yangzhao02**
This pull request is used to solve [When I use runc and criu 3.9, it crashed ](https://github.com/opencontainers/runc/issues/4381). When the criu version does not meet the requirements, execution will be terminated directly to prevent unpredictable behavior.

### CODE REVIEW

**Comment by lifubang** [comment]
Thanks, could you have a test case?

File: libcontainer/criu_linux.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func criuNsToKey(t configs.NamespaceType) string {
 
 func (c *Container) handleCheckpointingExternalNamespaces(rpcOpts *criurpc.CriuOpts, t configs.NamespaceType) error {
 	if !c.criuSupportsExtNS(t) {
-		return nil
+		return fmt.Errorf("criu version does not meet the requirements")
```
Comment (kolyshkin) [inline libcontainer/criu_linux.go:189]: Maybe something more specific, like
```go
return fmt.Errorf("criu lacks support for %s namespace (old criu version?)", t)
```

**Comment by kolyshkin** [comment]
> Thanks, could you have a test case?

It's hard to add a test as it would require installing some old criu version.

**Review by kolyshkin — COMMENTED** [review]
Should we do the same in `handleRestoringExternalNamespaces`?

File: libcontainer/criu_linux.go (hunk 1 above)
Comment (yangzhao02) [reply libcontainer/criu_linux.go:189]: Yes, that's better. I will change it later.

**Comment by yangzhao02** [comment]
> Should we do the same in `handleRestoringExternalNamespaces`?

Good idea, there is the same problem in `handleRestoringExternalNamespaces`. And I have do the same in `handleRestoringExternalNamespaces`.

**Comment by kolyshkin** [comment]
CI failures appear to be flakes (filed https://github.com/opencontainers/runc/issues/4437). Failed CI jobs restarted.

**Review by kolyshkin — CHANGES_REQUESTED** [review]
@yangzhao02 please avoid adding merge commits (use `git pull --rebase` to rebase your working tree)

**Comment by yangzhao02** [comment]
This pull request has been moved to https://github.com/opencontainers/runc/pull/4440

**Pull request closed without merging**